tfi_file.c

```diff
@@ -50,7 +50,23 @@ int tfi_file_load(struct tfi_file *tfi, uint8_t *data, size_t data_len) {
 }
 
 int tfi_file_save(struct tfi_file *f, int (*write_fn)(void *, size_t, void *), void *data_ptr) {
-	return -1;
+	uint8_t buf[42] = { 0 };
+	int bc = 0;
+	buf[bc++] = f->alg;
+	buf[bc++] = f->fb;
+	for(int i = 0; i < 4; i++) {
+		buf[bc++] = f->operators[i].mul;
+		buf[bc++] = f->operators[i].dt;
```

Comment:
Just one last thing, create a uint8_t * pointer and increment that instead of incrementing the index, `*(b++) = ...`